app/src/main/java/com/github/siela1915/bootcamp/Tools/LanguageFilter.java

```diff
@@ -0,0 +1,513 @@
+            "spooge",
+            "spread legs",
+            "spunk",
+            "strap on",
+            "strapon",
+            "strappado",
+            "strip club",
+            "style doggy",
+            "suck",
+            "suck my balls",
+            "sucks",
+            "suicide girls",
+            "sultry women",
+            "swastika",
+            "swinger",
+            "tainted love",
+            "taste my",
+            "tea bagging",
+            "threesome",
+            "throating",
+            "tied up",
+            "tight white",
+            "tits",
+            "titties",
+            "titty",
+            "tongue in a",
+            "topless",
+            "tosser",
+            "towelhead",
+            "tranny",
+            "tribadism",
+            "tub girl",
+            "tubgirl",
+            "tushy",
+            "twat",
+            "twink",
+            "twinkie",
+            "two girls one cup",
+            "undressing",
+            "upskirt",
+            "urethra play",
+            "urophilia",
+            "vagina",
+            "venus mound",
+            "vibrator",
+            "violet wand",
+            "vorarephilia",
+            "voyeur",
+            "vulva",
+            "wank",
+            "wetback",
+            "wet dream",
+            "white power",
+            "wrapping men",
+            "wrinkled starfish",
+            "yaoi",
+            "yellow showers",
+            "yiffy",
+            "zoophilia")
+    );
+    private final static int maxBadWordLength = 27;
+
+    /**
+     * Filters out rude words from a string. Replaces any rude words with stars ****
+     * @param input the string to filter.
+     * @return the filtered string.
+     * For a more extensive filter, see filterLangugage().
+     */
+    public static String filterSingleWords(String input) {
+        if (input == null || input.isEmpty()) {
+            return input;
+        }
+        //Don't forget to filter out any punctuation or other symbols
+        String s = filterLeetSpeach(input).toLowerCase().replaceAll("[^a-z]", " ");
+        Set<String> words = new HashSet<>(Arrays.asList(
+                s.split(" ")
+        ));
+        for (String word : words) {
+            if (badWords.contains(word)) {
+                String stars = stars(word.length());
+                // Add (?i) to remove case sensitivity.
+                for (String option : leetOptions(word)) {
+                    input = input.replaceAll("(?i)" + option, stars);
+                }
+            }
+        }
+        return input;
+    }
+
+    /**
+     * Filters out rude words and combinations of words that are considered rude and replaces
+     * the whole combination with stars ****
+     * @param input the string to filter.
+     * @return the filtered string.
+     * For simple rude word filtering see filterSingleWords().
+     */
+    public static String filterLanguage(String input) {
```

Comment:
the filterSingleWords and filterLanguage methods have some code repetition that could be modularized a bit to make the code more clean

Reply:
Only the first few lines of both methods are repeated. The rest is considerably different. I think there is no need to refactor this.